cylc/flow/scripts/cat_log.py

```diff
@@ -481,10 +490,13 @@ def main(
                 ),
                 reverse=True
             )
-            try:
-                log_file_path = Path(logs[rotation_number])
-            except IndexError:
-                raise InputError("max rotation %d" % (len(logs) - 1))
+            if logs:
+                try:
+                    log_file_path = Path(logs[rotation_number])
+                except IndexError:
+                    raise InputError("max rotation %d" % (len(logs) - 1))
```

Comment:
How about
```python
                    raise InputError(
                        f"--rotation {rotation_num} invalid "
                        f"(max value is {len(logs) - 1})"
                    )
```

Reply:
I think that's a legit change. I'll add it to the branch.